### PR TITLE
NMA-6594: Add SatsTabRow component and SatsTab

### PR DESCRIPTION
**Pull request by gurooerbech**
### What's new?
* Create a SatsTabRow
* Create a SatsTab (this doesn't do much but ensure that we use the correct M3 Tab for the tab row)

### CODE REVIEW

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt [hunk 1]
```diff
@@ -0,0 +1,93 @@
...
+@Composable
+fun SatsTabRow(
+    selectedTabIndex: Int,
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit,
+) {
+    CompositionLocalProvider(LocalUseMaterial3 provides true) {
+        TabRow(
+            selectedTabIndex = selectedTabIndex,
+            containerColor = Color.Transparent,
```
Comment (gurooerbech) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:28]: Not really sure about these colours yet as this component isn't in the figma library. But I've asked in the Design slack what colours this component should have 🙏

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt [hunk 2]
```diff
@@ -0,0 +1,93 @@
...
+import androidx.compose.runtime.CompositionLocalProvider
+import androidx.compose.runtime.remember
+import androidx.compose.ui.Modifier
+import androidx.compose.ui.graphics.Color
+import androidx.compose.ui.tooling.preview.Preview
+import com.sats.dna.LocalUseMaterial3
+import com.sats.dna.theme.SatsTheme
+
+@Composable
+fun SatsTabRow(
```
Comment (gurooerbech) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:20]: Idk how flexible we want this component to be?

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt [hunk 3]
```diff
@@ -0,0 +1,93 @@
...
+        enabled = enabled,
+        text = text,
+        icon = icon,
+        selectedContentColor = selectedContentColor,
+        unselectedContentColor = unselectedContentColor,
+        interactionSource = interactionSource,
+    )
+}
+
+@Preview
```
Comment (sindrenm) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:70]: Maybe `@PreviewLightDark`?

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt [hunk 4]
```diff
@@ -0,0 +1,93 @@
...
+import com.sats.dna.LocalUseMaterial3
+import com.sats.dna.theme.SatsTheme
+
+@Composable
+fun SatsTabRow(
+    selectedTabIndex: Int,
+    modifier: Modifier = Modifier,
+    content: @Composable () -> Unit,
+) {
+    CompositionLocalProvider(LocalUseMaterial3 provides true) {
```
Comment (sindrenm) [inline sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:25]: Does it make sense to instead just do this? It should probably always have the surface colors, one would think?

```suggestion
    SatsSurface(useMaterial3 = true) {
```

Comment (sindrenm) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:25]: Or maybe it's more correct to make it transparent, like I see you do below. 🤔

File: sample-app/src/main/kotlin/com/sats/dna/sample/screens/TabRowSampleScreen.kt [hunk 5]
```diff
@@ -0,0 +1,72 @@
...
+import androidx.compose.runtime.mutableStateOf
+import androidx.compose.runtime.remember
+import androidx.compose.runtime.setValue
+import androidx.compose.ui.Modifier
+import com.sats.dna.components.SatsTab
+import com.sats.dna.components.SatsTabRow
+
+data object TabRowSampleScreen : SampleScreen(
+    name = "TabRow",
+    route = "/components/tabrow",
```
Comment (sindrenm) [inline sample-app/src/main/kotlin/com/sats/dna/sample/screens/TabRowSampleScreen.kt:19]:
```suggestion
    route = "/components/tab-row",
```

**Review by sindrenm — APPROVED** [review]
👜

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt (hunk 4 above)
Comment (gurooerbech) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:25]: In the design it doesn't have a set background color afaict. Might be risky if we decide to put it somewhere where the background color doesn't fulfil double AA with the text color here.

[One of the instances where this is used](https://github.com/sats-group/member-app-android/blob/5d459a4cb56caf9d6238595aea8cb6c408065719/features/my-activity/src/main/kotlin/com/sats/features/myactivity/MyActivityTopAppBar.kt#L61), the background color follows that of the top app bar (with transitioning colors when scrolled). Which would not work with a set background color, so we could open this component to accept background colors, but then again we would have the same issue as with a transparent background unless we also open for changing content colours to match the background color which would then deviate from the colors set in the DS 🥴 

Tl;DR: We should maybe keep it transparent unless we want it to have totally customisable background and content colors.

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt (hunk 1 above)
Comment (gurooerbech) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:28]: Waiting for this before tweaking colors: https://github.com/sats-group/sats-dna-android/pull/437

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt (hunk 4 above)
Comment (gurooerbech) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:25]: I honestly don't know what's best there's no background color set in the DS but there are text / indicator colors. So idk if we should open the component up for both setting a background color and a content color. Or maybe just keep it transparent with fixed text and indicator colors 🤔

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt (hunk 1 above)
Comment (sindrenm) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:28]: This is available now. 👍

File: sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt (hunk 4 above)
Comment (sindrenm) [reply sats-dna/src/main/kotlin/com/sats/dna/components/SatsTabRow.kt:25]: Yeah, let's keep the background color transparent. That's what seems to be intended in the DS, too.

**Review by sindrenm — COMMENTED** [review]
Makes sense to me!
